### ISSUE TITLE
Build fails when vendoring

### ISSUE THREAD

**Issue by vladklokun** [kubescape/kubescape#832]
# Describe the bug
When vendoring dependencies, building Kubescape fails because pkg-config cannot find built `git2go` artifacts. `pkg-config` cannot find the artifacts because it is looking for them in the wrong directory: the `vendor` directory instead of the `git2go` directory.

# Environment
OS: Ubuntu 22.04 LTS
Version: current Git [master](https://github.com/kubescape/kubescape/tree/bb586892ba619993ef2510398cc2ad2f3f2f11e1). Kubescape version is irrelevant, since it is unable to build.

# Steps To Reproduce
1. Clone the repository
2. Run `make libgit2`
3. Run `go mod vendor`
4. Run `make build`
5. Observe the results

# Expected behavior
The build system is able to build the Kubescape executable and put it in the working directory.

# Actual Behavior
The build system fails with an error:
```
$ make build
# [SNIPPED FOR BREVITY]
k8s.io/client-go/kubernetes/typed/storage/v1
github.com/kubescape/opa-utils/reporthandling/results/v1/prioritization
github.com/kubescape/opa-utils/reporthandling/helpers/v1
github.com/libgit2/git2go/v33
# pkg-config --cflags --static -- /home/kvd/sandbox/kubescape/vendor/github.com/libgit2/git2go/v33/static-build/instal
l/lib/pkgconfig/libgit2.pc
Failed to open '/home/kvd/sandbox/kubescape/vendor/github.com/libgit2/git2go/v33/static-build/install/lib/pkgconfig/li
bgit2.pc': No such file or directory
No package '/home/kvd/sandbox/kubescape/vendor/github.com/libgit2/git2go/v33/static-build/install/lib/pkgconfig/libgit
2.pc' found
pkg-config: exit status 1
golang.org/x/mod/semver
# [SNIPPED FOR BREVITY]
```

# Additional context
I’ve run into this while I was trying to change the code of an upstream module—`opa-utils`. I wanted to change upstream code and adjust Kubescape accordingly.

While this use case is better served by using [Go workspaces](https://go.dev/doc/tutorial/workspaces), as I have discovered later, I think that vendoring modules should not break the build process.

A workaround that helped me with vendoring before I moved on to Go workspaces was:
1. Bulid the git2go: `make libgit2`
2. Vendor the packages: `go mod vendor`
3. Copy the bulit `libgit2` directory to the `vendor` directory: `cp -r ./git2go ./vendor/github.com/kubescape/libgit2/git2go/v33`
4. Run the build: `make build`

**Comment by Meyazhagan** [comment]
@vladklokun  I too got similar error message while `make build`. After Copying the bulit `libgit2` directory to the `vendor`, build succeeds. Shall I debug it !

**Comment by Meyazhagan** [comment]
@vladklokun In `go.mod` file, I saw `replace github.com/libgit2/git2go/v33 => ./git2go` was this causing the issue

**Comment by Meyazhagan** [comment]
@dwertent shall I work on this !

**Comment by Aman123lug** [comment]
@Meyazhagan can i work with you man?

**Comment by 0xt3j4s** [comment]
Hey @vladklokun and @Meyazhagan,
Since there have been no updates on this issue for a long time now, I tried to tinker a bit and followed the steps to reproduce the bug. The build (`make build`) fails with an error as given below:
```bash
make: *** [Makefile:25: build] Error 2
```

To get the context and more details, I tried the following:
```bash
$ make         
git submodule update --init --recursive
cd git2go; make install-static
make[1]: Entering directory '/home/tejas/Projects/Kubescape/kubescape/git2go'
go install --tags "static" ./...
make[1]: Leaving directory '/home/tejas/Projects/Kubescape/kubescape/git2go'
go build -v -tags="gitenabled,static" .
github.com/libgit2/git2go/v33
# pkg-config --cflags --static -- /home/tejas/Projects/Kubescape/kubescape/vendor/github.com/libgit2/git2go/v33/static-build/install/lib/pkgconfig/libgit2.pc
Package vendor/github.com/libgit2/git2go/v33/static-build/install/lib/pkgconfig/libgit2.pc was not found in the pkg-config search path.
Perhaps you should add the directory containing `/home/tejas/Projects/Kubescape/kubescape/vendor/github.com/libgit2/git2go/v33/static-build/install/lib/pkgconfig/libgit2.pc.pc'
to the PKG_CONFIG_PATH environment variable
Package '/home/tejas/Projects/Kubescape/kubescape/vendor/github.com/libgit2/git2go/v33/static-build/install/lib/pkgconfig/libgit2.pc', required by 'virtual:world', not found
pkg-config: exit status 1
make: *** [Makefile:25: build] Error 2
```

I think there is some issue with git2go in the dependencies, i.e., in the `go.mod`. For this, I also referred to @amirmalka's [comment](https://github.com/kubescape/kubescape/issues/1014#issuecomment-1376933521) regarding the removal of git2go and updating of the scanning logic of a repository.


I am interested in working on this issue. How should I proceed further?

**Comment by 0xt3j4s** [comment]
Sure, I'll try to figure out why `pkg-config` is looking in the vendor directory and also read the documentation of `git2go`. 
Thank you for the insight

**Comment by dwertent** [comment]
Resolved in release [v3.0.4](https://github.com/kubescape/kubescape/releases/tag/v3.0.4)
